### PR TITLE
Convert `org.pkl.core` POJOs to `record` classes

### PR DESCRIPTION
**Pull request by KushalP**

### CODE REVIEW

File: pkl-core/src/main/java/org/pkl/core/Platform.java [hunk 1]
```diff
@@ -17,14 +17,24 @@
...
 
 /**
  * Information about the Pkl release that the current program runs on. This class is the Java
  * equivalent of standard library module {@code pkl.platform}.
+ *
+ * @param language the language implementation of this platform.
+ * @param runtime the language runtime of this platform.
+ * @param virtualMachine the virtual machine of this platform.
+ * @param operatingSystem the operating system of this platform.
+ * @param processor the processor of this platform.
```
Comment (holzensp) [inline pkl-core/src/main/java/org/pkl/core/Platform.java:30]: Not the _most_ elucidating documentation ;) I'd leave it out, tbh (also makes it easier to spot it isn't _really_ documented yet)

**Review by holzensp — APPROVED** [review]
Convenient that most already had getters named in record style

File: pkl-core/src/main/java/org/pkl/core/Platform.java [hunk 2]
```diff
@@ -17,14 +17,18 @@
...
  * Information about the Pkl release that the current program runs on. This class is the Java
  * equivalent of standard library module {@code pkl.platform}.
  */
-public final class Platform {
+public record Platform(
+    org.pkl.core.Platform.Language language,
+    org.pkl.core.Platform.Runtime runtime,
+    org.pkl.core.Platform.VirtualMachine virtualMachine,
+    org.pkl.core.Platform.OperatingSystem operatingSystem,
+    org.pkl.core.Platform.Processor processor) {
```
Comment (bioball) [inline pkl-core/src/main/java/org/pkl/core/Platform.java:31]:
```suggestion
    Language language,
    Runtime runtime,
    VirtualMachine virtualMachine,
    OperatingSystem operatingSystem,
    Processor processor) {
```

**Review by bioball — CHANGES_REQUESTED** [review]

**Review by bioball — APPROVED** [review]
Love the deleted line count!

**Pull request merged**